### ISSUE TITLE
Add CQL_Filter to optional WMS params

### ISSUE THREAD

**Issue by zachrouseFE** [jupyter-widgets/ipyleaflet#264]
The CQL_Filter is a handy part of the WMS spec that let's us filter the features that get rendered in WMS requests.  https://docs.geoserver.org/stable/en/user/tutorials/cql/cql_tutorial.html

I think it should just drop in like this:

https://github.com/jupyter-widgets/ipyleaflet/blob/master/js/src/jupyter-leaflet.js#L1274
`cql_filter : null,`

https://github.com/jupyter-widgets/ipyleaflet/blob/master/ipyleaflet/leaflet.py#L255
`cql_filter = Unicode(allow_none=True, default_value=None).tag(sync=True, o=True)`

**Comment by martinRenou** [comment]
It won't work if you just add a `cql_filter` attribute like that, it would if `cql_filter` was an option for the `Tilelayer.WMS` class from LeafletJS, but it's not: https://leafletjs.com/reference-1.4.0.html#tilelayer-wms.

I am not able to find any mention to a `cql_filter` attribute in the LeafletJS code, I only found [an old issue](https://github.com/Leaflet/Leaflet/issues/708) mentionning it. I am not sure it is supported by LeafletJS, and if it was I guess it is not anymore.

**Comment by zachrouseFE** [comment]
@martinRenou Thanks.  I am sneaking it in to the wms requests with this approach but it feels hacky if it is not a supported attribute.

**Comment by martinRenou** [comment]
Yeah sorry, I don't really have another option for you. I guess we can add your little hack on the Python side of ipyleaflet. We could add the `cql_filter` attribute to `WMSTileLayer` and every time this attribute changes we update the url of the layer, just by formating the string. Do you think it could work?

**Comment by zachrouseFE** [comment]
That would cover it.  There has to be a new request to grab tiles that use the updated cql_filter anyways.

**Comment by martinRenou** [comment]
Yep, maybe there is a way to enforce the reload of tiles with LeafletJS. Would you like to try adding it to ipyleaflet?

**Comment by martinRenou** [comment]
Yeah we can listen on the javascript side for changes on the url and use [seturl](https://leafletjs.com/reference-1.4.0.html#tilelayer-seturl) which will redraw tiles :)

**Comment by bhogan-mitre** [comment]
Note that `cql_filter` is supported implicitly by LeafletJS:
> If any custom options not documented here are used, they will be sent to the WMS server as extra parameters in each request URL. This can be useful for non-standard vendor WMS parameters.
via https://leafletjs.com/reference-1.5.0.html#tilelayer-wms-option

So a variant of the `TileLayer.WMS` [example](https://leafletjs.com/reference-1.5.0.html#tilelayer-wms) that includes this would be:
```
var nexrad = L.tileLayer.wms("http://mesonet.agron.iastate.edu/cgi-bin/wms/nexrad/n0r.cgi", {
    layers: 'nexrad-n0r-900913',
    format: 'image/png',
    transparent: true,
    cql_filter: 'INCLUDE',
    attribution: "Weather data © 2012 IEM Nexrad"
});
```

Specifying `INCLUDE` as the default means that no filters are applied.

Would that be straight-forward to implement @martinRenou ?
